### PR TITLE
:sparkles: Feedback functionality

### PR DESCRIPTION
**Pull request by soundsmitten**
Feedback button in navigation bar and functionality using a generic form, per issue #136

### CODE REVIEW

File: teferi/ViewControllers/MainViewController.swift [hunk 1]
```diff
@@ -249,4 +255,53 @@ class MainViewController : UIViewController
...
+        dismiss(animated: true)
+    }
+    
+    //Setup compose email form
+    func composeEmail()
+    {
+        //Check if email is configured on device
+        guard MFMailComposeViewController.canSendMail() else
+        {
+            let alert = UIAlertController(title: "Can't send email.", message: "Mail is not configured on his device. Either set it up in \"Settings -> Mail, Contacts, Calendars\" or email support@toggl.com using your preferred email client", preferredStyle: .alert)
```
Comment (willsb) [inline teferi/ViewControllers/MainViewController.swift]: Did you check this copy with Heidi? The "his device" part sounds kinda off :P

Comment (soundsmitten) [inline teferi/ViewControllers/MainViewController.swift]: No I haven't. I'll talk to her.

**Comment by soundsmitten** [comment]
Wording for the prompts is now corrected.

**Comment by willsb** [comment]
Nice work on this one 💃

**Comment by amulware** [comment]
What is the status of this? Why was it approved if significant changes were expected? Is it now done?

File: teferi/Services/Protocols/FeedbackService.swift [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import UIKit
+
+protocol FeedbackService
+{
+    func composeEmail(recipients: [String], subject: String, body: String, logURL: URL?, parentViewController viewController: UIViewController)
```
Comment (amulware) [inline teferi/Services/Protocols/FeedbackService.swift]: The recipients, subject and body are specific to the email-feedback implementation of this service and should not be part of the protocol.

**Comment by soundsmitten** [comment]
For the bug @willsb mentioned:

> Scroll to a day in the past
> Click the send feedback button
> Cancel it and delete the draft
> It will scroll back to current day, but the header will still display the older day. 
> I believe it shouldn't scroll back to the current date at all.

Located the issue here. PagerViewController's `viewWillAppear` is called when the email draft closes. In `viewWillAppear`, the date displayed in the TimeLineViewController is always set to the current `Date()`. Not sure about where to fix this without getting really messy/hackish  (likely with a boolean somewhere).

**Comment by amulware** [comment]
It seems like a bug in the first place that the pager can switch days without switching title. That should be fixed. In addition, the original design said that the pager only switches back to 'today' if it was not visible for at least an hour (it was meant to have the app switch to today when the user had it in the background over night, and then started it the next day). I'm not sure that is necessarily the correct fix for this though. Ideally it should only consider switching day in the case where the app was opened from the background (maybe have the pager subscribe to the appstate's active state).

Just some thoughts. ¯\_(ツ)_/¯

**Comment by amulware** [comment]
While it is a solution, I don't like that latest commit. It makes no sense to have that dependency. The solution should be more general. Otherwise we'll have to inject even more things in the future, if we add more features like this.

**Comment by soundsmitten** [comment]
Last change is more generic.

File: teferi/ViewControllers/PagerViewController.swift [hunk 3]
```diff
@@ -78,11 +76,13 @@ class PagerViewController : UIPageViewController, UIPageViewControllerDataSource
             .addDisposableTo(disposeBag!)
         
         //Check if the feedback UI has closed
-        if self.feedbackService.hasStartedFeedback == false {
-            self.initCurrentDateViewController()
-        } else {
-            self.feedbackService.hasStartedFeedback = false
+        guard !self.feedbackUIClosing else
```
Comment (amulware) [inline teferi/ViewControllers/PagerViewController.swift]: I'd like this refactored into an if-statement.

File: teferi/Services/Protocols/MailFeedbackService.swift [hunk 4]
```diff
@@ -5,14 +5,13 @@ import MessageUI
...
     //MARK: Fields
-    private let recipients: [String]
-    private let subject: String
-    private let body: String
-    
-    private var feedbackStarted: Bool = false
+    private let recipients : [String]
+    private let subject : String
+    private let body : String
+    private var completionHandler: () -> ()
```
Comment (amulware) [inline teferi/Services/Protocols/MailFeedbackService.swift]: There is no need for this property.

**Comment by amulware** [comment]
Superseded by #153

**Pull request closed without merging**